ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -35,32 +35,32 @@ class DataCollectionViewModel @Inject internal constructor(
     private val submissionRepository: SubmissionRepository,
     private val locationOfInterestHelper: LocationOfInterestHelper
 ) : AbstractViewModel() {
-    val submission: @Hot(replays = true) LiveData<Loadable<Submission>>
+    val submission: @Hot(replays = true) LiveData<Submission?>
     val jobName: @Hot(replays = true) LiveData<String>
     val loiName: @Hot(replays = true) LiveData<String>
 
     private val argsProcessor: @Hot(replays = true) FlowableProcessor<DataCollectionFragmentArgs> =
         BehaviorProcessor.create()
 
     init {
-        val submissionStream: Flowable<Loadable<Submission>> =
+        val submissionStream: Flowable<Submission?> =
             argsProcessor.switchMapSingle { args ->
                 submissionRepository.createSubmission(
                     args.surveyId, args.locationOfInterestId, args.submissionId
-                ).map { Loadable.loaded(it) }.onErrorReturn { Loadable.error(it) }
+                ).map { it }.onErrorReturn { null }
             }
 
         submission = LiveDataReactiveStreams.fromPublisher(submissionStream)
 
-        jobName = LiveDataReactiveStreams.fromPublisher(submissionStream.map { submission ->
-            submission.value().map { it.locationOfInterest.job.name }.orElse("")
+        jobName = LiveDataReactiveStreams.fromPublisher(submissionStream.map {
+            it.locationOfInterest.job.name ?: ""
```

Comment:
Out of curiosity, this LiveData is used only by the text field in UI, right? Isn't that already null safe?

Reply:
I think for loiName it's safe because it's used in the TwoLineToolbar which accepts a `String?` in `setTitle()`, however the jobName is used in a `TextView` which doesn't accept null values, so I think I'd rather just keep the strings non-null here.